src/sync/submitters/telemetrySubmitter.ts

```diff
@@ -38,6 +39,25 @@ function getRedundantActiveFactories() {
   }, 0);
 }
 
+function getTelemetryFlagSetsStats(splitFiltersValidation: ISplitFiltersValidation) {
```

Comment:
Having an internal `originalSets` array is unnecessary. We can reduce code by counting the number of total set filters in a reducer:

```
function getTelemetryFlagSetsStats(splitFiltersValidation: ISplitFiltersValidation) {
  const flagSetsTotal = splitFiltersValidation.validFilters.reduce((accum: number, filter: SplitIO.SplitFilter) => {
    return accum + (filter.type === 'bySet' ? filter.values.length : 0);
  }, 0);

  const flagSetsValid = splitFiltersValidation.groupedFilters.bySet.length;
  const flagSetsIgnored = flagSetsTotal - flagSetsValid;
  return { flagSetsTotal, flagSetsIgnored };
}
```

Reply:
great catch doc! thank you!